backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/database/jpa/DatabaseDocumentationUnitRepository.java

```diff
@@ -46,16 +47,22 @@ status.publicationStatus IN (de.bund.digitalservice.ris.caselaw.domain.Publicati
       )
     )
    AND (:withErrorOnly = FALSE OR documentationUnit.documentationOffice.id = :documentationOfficeId AND documentationUnit.status.withError = TRUE)
-ORDER BY documentationUnit.decisionDate DESC NULLS LAST
+ORDER BY
+ CASE
+  WHEN (:onlyScheduled IS TRUE OR cast(:publicationDate as date) IS NOT NULL) THEN documentationUnit.scheduledPublicationDate ELSE documentationUnit.decisionDate END DESC NULLS LAST
 """;
 
+  // TODO: Order by CASE WHEN
+
   @Query(
       value =
           """
   SELECT documentationUnit FROM DocumentationUnitDTO documentationUnit
   LEFT JOIN documentationUnit.court court
   LEFT JOIN documentationUnit.status status
   WHERE
+   (cast(:publicationDate as date) IS NULL OR documentationUnit.lastPublicationDate = :publicationDate OR documentationUnit.scheduledPublicationDate = :publicationDate) AND
```

Comment:
`publicationDate` (-> Date input from the search param) should be a `LocalDate` and instead of a `LocalDateTime`. Then we need to check for the date only ignoring time (from 00:00 - 23:59).